jenkins/jobs/e2e_features_test.pipeline

```diff
@@ -0,0 +1,120 @@
+import java.text.SimpleDateFormat
+
+ci_git_credential_id = "metal3-jenkins-github-username-token"
+
+// 10 minutes
+def CLEAN_TIMEOUT = 600
+// default 3h for other non-pivot feature tests
+def TIMEOUT = 10800
+
+script {
+  UPDATED_REPO = "https://github.com/${env.REPO_OWNER}/${env.REPO_NAME}.git"
+  echo "Test triggered from ${UPDATED_REPO}"
+  ci_git_url = "https://github.com/metal3-io/project-infra.git"
+
+  if ("${env.REPO_OWNER}" == 'metal3-io' && "${env.REPO_NAME}" == 'project-infra') {
+    ci_git_branch = (env.PULL_PULL_SHA) ?: 'main'
+    ci_git_base = (env.PULL_BASE_REF) ?: 'main'
+    // Fetch the base branch and the ci_git_branch when running on project-infra PR
+    refspec = '+refs/heads/' + ci_git_base + ':refs/remotes/origin/' + ci_git_base + ' ' + ci_git_branch
+  } else {
+    ci_git_branch = 'main'
+    refspec = '+refs/heads/*:refs/remotes/origin/*'
+  }
+  echo "Checkout ${ci_git_url} branch ${ci_git_branch}"
+  if ("${GINKGO_FOCUS}" == 'pivoting') {
+    BUILD_TAG = "${env.BUILD_TAG}-pivoting-based"
+    TIMEOUT = 18000 // 5h for node reuse
+  } else if ("${GINKGO_FOCUS}" == 'remediation') {
+    BUILD_TAG = "${env.BUILD_TAG}-remediation-based"
+    TIMEOUT = 18000 // 5h for remediation
+  } else {
+    BUILD_TAG = "${env.BUILD_TAG}-other-features"
+    GINKGO_SKIP = "pivoting remediation" // Allow non pivoting features
+  }
+  agent_label="metal3ci-large-${IMAGE_OS}"
+}
+
+pipeline {
+  agent { label agent_label }
+  environment {
+    REPO_ORG = "${env.REPO_OWNER}"
+    REPO_NAME = "${env.REPO_NAME}"
+    UPDATED_REPO = "${UPDATED_REPO}"
+    REPO_BRANCH = "${env.PULL_BASE_REF}"
+    UPDATED_BRANCH = "${env.PULL_PULL_SHA}"
+    BUILD_TAG = "${BUILD_TAG}"
+    PR_ID = "${env.PULL_NUMBER}"
+    IMAGE_OS = "${IMAGE_OS}"
+    CAPI_VERSION = "${CAPI_VERSION}"
+    CAPM3_VERSION = "${CAPM3_VERSION}"
+    CAPM3RELEASEBRANCH = "${capm3_release_branch}"
+    BMORELEASEBRANCH = "${bmo_release_branch}"
+    TARGET_NODE_MEMORY = "${TARGET_NODE_MEMORY}"
+    NUM_NODES = 4
+    TESTS_FOR = "${TESTS_FOR}"
+    GINKGO_FOCUS = "${GINKGO_FOCUS}"
+    GINKGO_SKIP = "${GINKGO_SKIP}"
```

Comment:
I need suggestion: Do I need to add `GINKGO_SKIP` variable in pipeline? JJB is not going to pass this var but this pipeline will add the value

Reply:
If JJB is not passing it I don't see much use of this var in this form. This can be either removed or if you need you can hardcode some value here e.g. GINKGO_SKIP="whatever"

Reply:
it is necessary to pass GINKGO_SKIP when GINKGO_FOCUS=features. 
You can add condition here in the pipeline  GINKGO_FOCUS == features, GINKGO_SKIP== pivoting  remediation.

Reply:
Okay you are giving a value here https://github.com/metal3-io/project-infra/pull/811/files#diff-7654f3f0b93e2e0e5eac4e5b579b787cc7644bd0ab047192211c8dbc5d785cfeR33

Reply:
After discussion, we can see that this var is needed